### DISCUSSION TITLE
Open customizations when selecting applicable mod

### DISCUSSION THREAD

**Discussion by ackhack** [ppy/osu#17348] [Ideas] [answered]
I added the Functionality to automatically show the `ModSettingsContainer` when a mod gets selected that can be configured and binded it to a new `OsuSetting` called `AutoShowModSettings`.
The Implementation can be found [here](https://github.com/ackhack/osu/commit/671bc95c0e79888895fa55ac75bbc30f6cbbccc7).
Can this be PRed into master?

**Comment by bdach** [comment]
i'm personally opposed to this as usual when new settings are suggested. this should happen either always or as currently done, only when a mod *requires* configuration to do anything.

also note that the mod overlay is actively undergoing a redesign.

**Comment by ackhack** [comment]
I can see why people wouldnt want it to show everytime, because it can be in the way in smaller screens and be confusing to newer people. But thats why i put the functionality behind a setting, that defaults to false, so people like me with big screens dont have to click the customizations everytime i want to have a different speed, while others dont have to bother with that.

**Reply by bdach** [reply]
not everything should be a setting. adding a setting for everything is a surefire way to make the settings menu unusable for everybody.

**Comment by peppy** [comment, marked as answer]
We can do this in a better way with the new mod select overlay. It should not be a toggle.
